flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/slot/TaskSlotTable.java

```diff
@@ -448,7 +448,7 @@ public boolean addTask(Task task) throws SlotNotFoundException, SlotNotActiveExc
 				throw new SlotNotActiveException(task.getJobID(), task.getAllocationId());
 			}
 		} else {
-			throw new SlotNotFoundException(taskSlot.getAllocationId());
+			throw new SlotNotFoundException(task.getAllocationId());
```

Comment:
doesn't seem to be equivalent?

Reply:
this is in the else branch of `taskSlot != null`, as in it would always fail with a `NullPointerException`. @tillrohrmann

Reply:
Yes, this is a bug. Thanks for fixing it.